src/sidebar/services/groups.js

```diff
@@ -378,8 +378,9 @@ export class GroupsService {
       userGroups.find(g => g.id === id || g.groupid === id) ||
       tryFetchGroup(id);
 
-    const groups = (await Promise.all(groupIds.map(getGroup))).filter(
-      g => g !== null
+    const groupResults = await Promise.all(groupIds.map(getGroup));
+    const groups = /** @type {Group[]} */ (
+      groupResults.filter(g => g !== null)
     );
```

Comment:
This code fetches group details for a list of group IDs. This change is just asserting that we discard any `null` results from `getGroup` where the group could not be found.

Reply:
The change makes it easier to read, too. I'm all about chaining and smart use of functional tactics, but... :)